### PR TITLE
2960 replace bash script task file local dev

### PR DESCRIPTION
**Pull request by raftmsohani**
## Summary of Changes
_Provide a brief summary of changes_
Pull request closes #2960_

## How to Test

Run ```task help``` in project's root directory which will list all the available command. Feel free to run any of the command and test.

## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
+ [ ] [**_insert ACs here_**]
+ [ ] **`lfrohlich`** and/or **`adpennington`**  confirmed that ACs are met.

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
  + [ ] If this PR introduces backend code changes, are they meaningfully tested?
  + [ ] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [ ] Frontend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)
  + [ ] Backend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)

+ [x] Are backend code style checks passing on CircleCI?
+ [x] Are frontend code style checks passing on CircleCI?
+ [ ] Are code maintainability principles being followed?

### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with `iamjolly` and `ttran-hub` using Accessibility Insights reveal any errors introduced in this PR?

### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

+ [x] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

+ [ ] Does this PR provide background for why coding decisions were made?
+ [ ] If this PR introduces backend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces frontend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?
+ [ ] Can reviewer explain and take ownership of these elements presented in this code review?

### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

+ [ ] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any new security issues?
+ [ ] If new issues detected, is investigation and/or remediation plan documented? 

### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
+ [ ] the purpose of the research
+ [ ] methods used to conduct the research 
+ [ ] who participated in the research
+ [ ] what was tested and how
+ [ ] impact of research on TDP
+ [ ] (_if applicable_) final design mockups produced for TDP development

### CODE REVIEW

File: Taskfile.yml [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+  frontend-init:
+    desc: Initialize the frontend project
+    dir: tdrs-frontend
+    cmds:
+      - docker-compose -f docker-compose.yml up -d --build
+      - docker-compose -f docker-compose.yml exec tdp-frontend sh -c "apk add nodejs npm"
+      - docker-compose -f docker-compose.yml exec tdp-frontend sh -c "npm install"
+      - docker-compose -f docker-compose.yml down
+
+  frontend-test:
```
Comment (raftmsohani) [inline Taskfile.yml:148]: I cannot get it to exit when it's done, nor accepts input. have to use ctrl+c to exit.

File: Taskfile.yml [hunk 2]
```diff
@@ -0,0 +1,166 @@
+version: '3'
```
Comment (raftmsohani) [inline Taskfile.yml:1]: TODO: add installation to docs: https://taskfile.dev/installation/

File: Taskfile.yml [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+
+tasks:
+
+  create-network:
+    desc: Create the external network
+    cmds:
+      - docker network create external-net
+
+  init-backend:
```
Comment (andrew-jameson) [inline Taskfile.yml:10]: After running `task clean`, would I always need to run `init-backend` before `backend-up`?

**Comment by andrew-jameson** [comment]
Shouldn't PR also remove the `commands.sh` file? Or is it missing some of those commands yet?

File: Taskfile.yml (hunk 1 above)
Comment (raftmsohani) [reply Taskfile.yml:148]: run in the background, and then exec terminal into the docker instance

Comment (raftmsohani) [reply Taskfile.yml:148]: try: npm test instead of npm run test

Comment (raftmsohani) [reply Taskfile.yml:148]: another idea is to : "test": "CI=true react-scripts test"

File: Taskfile.yml (hunk 3 above)
Comment (raftmsohani) [reply Taskfile.yml:10]: If you need to do migrations, then yes

File: Taskfile.yml [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+    dir: tdrs-frontend
+    cmds:
+      - docker-compose -f docker-compose.yml exec tdp-frontend sh
+
+  frontend-pip-lock:
+    desc: Lock the pip dependencies
+    dir: tdrs-frontend
+    cmds:
+      - task: frontend-up
+      - docker-compose -f docker-compose.yml exec tdp-frontend sh -c "pipenv lock"
```
Comment (elipe17) [inline Taskfile.yml]: pipenv shouldnt exist in this container right?

Comment (raftmsohani) [reply Taskfile.yml]: oops, this is a typo, from older merge

**Review by andrew-jameson — APPROVED** [review]
works on my machine, thank you so much for this, Mo!!

**Review by elipe17 — APPROVED** [review]

**Pull request merged**